src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -54,6 +54,11 @@ public PythonEngineVersion Engine
                 if (!Enum.TryParse(engine, out PythonEngineVersion engineVersion) ||
                     engineVersion == PythonEngineVersion.Unspecified)
                 {
+                    //if this is a valid dynamically loaded engine, return unknown, and serialize the name.
+                    if (PythonEngineManager.Instance.AvailableEngines.Any(x=>x.Name == engine))
+                    {
+                        return PythonEngineVersion.Unknown;
```

Comment:
why isn't PythonEngineVersion.Unspecified good enough to be returned ?

Reply:
@pinzart90 - two reasons IMO:
1. If the version is unspecified it makes sense to me that the default version should kick in.
2. Unspecified is not the same as Unknown. In this case the version is definitely specified, it's just not a valid value in the enum.

Reply:
So all this is to enhance the user experience for an obsoleted API? Is it still used by our integrators ?

Reply:
hah, not quite, I don't know who uses it, and I don't know when it will disappear, and IMO it is about enhancing correctness not user experience - not collapsing two different states into one.

Let us remove this enum and never mention it again, but until then, I don't want to break python node initialization and I also want dynamic engines to serialize correctly.